sdk/eventhub/Azure.Messaging.EventHubs.CheckpointStore.Blob/tests/Infrastructure/StorageScope.cs

```diff
@@ -0,0 +1,190 @@
+        ///   Initializes a new instance of the <see cref="StorageScope"/> class.
+        /// </summary>
+        ///
+        /// <param name="containerName">The name of the blob container that was created.</param>
+        ///
+        private StorageScope(string containerName)
+        {
+            ContainerName = containerName;
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to remove the dynamically created blob container.
+        /// </summary>
+        ///
+        public async ValueTask DisposeAsync()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            var resourceGroup = TestEnvironment.EventHubsResourceGroup;
+            var storageAccount = StorageTestEnvironment.StorageAccountName;
+            var token = await ResourceManager.AquireManagementTokenAsync();
+            var client = new StorageManagementClient(new TokenCredentials(token)) { SubscriptionId = TestEnvironment.EventHubsSubscription };
+
+            try
+            {
+                await ResourceManager.CreateRetryPolicy().ExecuteAsync(() => client.BlobContainers.DeleteAsync(resourceGroup, storageAccount, ContainerName));
+            }
+            catch
+            {
+                // This should not be considered a critical failure that results in a test failure.  Due
+                // to ARM being temperamental, some management operations may be rejected.  Throwing here
+                // does not help to ensure resource cleanup only flags the test itself as a failure.
+                //
+                // If a blob container fails to be deleted, removing of the associated storage account at the end
+                // of the test run will also remove the orphan.
+            }
+            finally
+            {
+                client?.Dispose();
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to create a new blob container instance with a dynamically assigned unique name.
+        /// </summary>
+        ///
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        /// <returns>The <see cref="StorageScope" /> in which the test should be executed.</returns>
+        ///
+        public static async Task<StorageScope> CreateAsync([CallerMemberName] string caller = "")
+        {
+            caller = InvalidContainerCharactersExpression.Replace(caller.ToLowerInvariant(), String.Empty);
+            caller = (caller.Length < 16) ? caller : caller.Substring(0, 15);
+
+            var resourceGroup = TestEnvironment.EventHubsResourceGroup;
+            var storageAccount = StorageTestEnvironment.StorageAccountName;
+            var token = await ResourceManager.AquireManagementTokenAsync();
+
+            string CreateName() => $"{ Guid.NewGuid().ToString("D").Substring(0, 13) }-{ caller }";
```

Comment:
Why a function rather than a string?

Reply:
There were some cases where ARM would throw a failure though creation was successful, we would retry and then be blocked on a 409 because the name was used.   The function is creating a new name for each retry pass, which improved stability of the nightly runs quite a bit.

ARM was not created with ephemeral resources in mind and gets very temperamental with frequent management operations, such as our tests running in parallel and creating a bunch of requests.